mgmtfn/mesosplugin/netDriver.go

```diff
@@ -0,0 +1,323 @@
+/***
+Copyright 2016 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mesosplugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/contiv/netplugin/drivers"
+	"github.com/contiv/netplugin/mgmtfn/mesosplugin/api"
+	"github.com/contiv/netplugin/netlib"
+	"github.com/contiv/netplugin/netmaster/intent"
+	"github.com/contiv/netplugin/netmaster/master"
+	"github.com/contiv/netplugin/netmaster/mastercfg"
+	"github.com/contiv/netplugin/netplugin/cluster"
+	"github.com/contiv/netplugin/utils"
+	"github.com/contiv/netplugin/utils/netutils"
+	"github.com/jainvipin/bitset"
+	"github.com/vishvananda/netlink"
+)
+
+const defaultTenantName = "default"
+
+type endpointState struct {
+	endpointRequest master.CreateEndpointRequest
+	endpoint        *drivers.OvsOperEndpointState
+	network         *mastercfg.CfgNetworkState
+	vethMasterName  string
+	vethIP          string
+}
+
+type endpoints struct {
+	epStates map[string]endpointState
```

Comment:
@unclejack Can you please open a github issue to persist this state? It'll be easy to keep track that way..